lib/charms/opensearch/v0/opensearch_backups.py

```
@@ -220,48 +226,50 @@ def __init__(self, charm: "OpenSearchBaseCharm", relation_name: str = PeerCluste
-        """Returns the output of OpenSearchDistribution.request() or throws an error.
-
-        Request method can return one of many: Union[Dict[str, any], List[any], int]
-        and raise multiple types of errors.
-
-        If int is returned, then throws an exception informing the HTTP request failed.
-        If the request fails, returns the error text or None if only status code is found.
-
-        Raises:
-          - ValueError
-        """
-        if "retries" not in kwargs.keys():
-            kwargs["retries"] = 6
-        if "timeout" not in kwargs.keys():
-            kwargs["timeout"] = 10
-        # We are interested to see the entire response
-        kwargs["resp_status_code"] = False
-        try:
-            result = self.charm.opensearch.request(*args, **kwargs)
-        except OpenSearchHttpError as e:
-            return e.response_body if e.response_body else None
-        return result if isinstance(result, dict) else None
-
     def _is_restore_in_progress(self) -> bool:
         """Checks if the restore is currently in progress.
 
         Two options:
          1) no restore requested: return False
          2) check for each index shard: for all type=SNAPSHOT and stage=DONE, return False.
         """
-        indices_status = self._request("GET", "/_recovery?human") or {}
+        try:
+            indices_status = (
+                self.charm.opensearch.request(
+                    "GET",
+                    "/_recovery?human",
+                    retries=6,
+                    timeout=10,
+                )
+                or {}
+            )
+        except OpenSearchHttpError:
+            # Defaults to True if we have a failure, to avoid any actions due to
+            # intermittent connection issues.
+            logger.warning(
+                "_is_restore_in_progress: failed to get indices status"
+                " - assuming restore is in progress"
+            )
+            return True
```

I'm not sure I understand this path, why return true and assume a restore is in progress when it may not be, instead of let it crash?

---

I kept the exception handling so we can add an error log here. I will reraise this exception however.